src/main/java/graphql/execution/ExecutionContext.java

```diff
@@ -170,6 +170,34 @@ public ValueUnboxer getValueUnboxer() {
         return valueUnboxer;
     }
 
+    /**
+     * @return true if the current operation is a Query
+     */
+    public boolean isQueryOperation() {
+        return isOpType(OperationDefinition.Operation.QUERY);
+    }
+
+    /**
+     * @return true if the current operation is a Mutation
+     */
+    public boolean isMutationOperation() {
+        return isOpType(OperationDefinition.Operation.MUTATION);
+    }
+
+    /**
+     * @return true if the current operation is a Subscription
+     */
+    public boolean isSubscriptionOperation() {
+        return isOpType(OperationDefinition.Operation.SUBSCRIPTION);
+    }
```

Comment:
I hav long wanted this.  Checking operationDefinition.getOperation() is a PITA

Reply:
Nice